frontend/models/product-list/index.ts

```diff
@@ -74,6 +82,48 @@ export async function findProductList(
    };
 }
 
+type DeviceWiki = Record<string, any>;
+
+async function fetchDeviceWiki(
+   deviceTitle: string
+): Promise<DeviceWiki | null> {
+   const deviceHandle = getDeviceHandle(deviceTitle);
+   try {
+      const response = await fetch(
+         `${IFIXIT_ORIGIN}/api/2.0/wikis/CATEGORY/${deviceHandle}`
+      );
+      const payload = await response.json();
+      return payload;
+   } catch (error) {
+      return null;
+   }
+}
+
+function getDeviceImage(deviceWiki: DeviceWiki): ProductListImage | null {
+   return deviceWiki.image?.original == null
+      ? null
+      : {
+           url: deviceWiki.image.original,
+           alternativeText: null,
+        };
+}
+
+function getChildDeviceImage(
+   deviceWiki: DeviceWiki,
+   childDeviceTitle: string
+): ProductListImage | null {
+   const child = deviceWiki.children?.find(
+      (c: any) => c.title === childDeviceTitle
+   );
+   if (child?.image?.original) {
+      return {
+         url: child.image.original,
+         alternativeText: null,
+      };
+   }
+   return null;
```

Comment:
It might be nice to reuse `getDeviceImage` here, but this is fine, not worth changing.

Reply:
I like your suggestion. The reason I haven't done it this way is that I would have to change the assumption about the typescript type that gets passed to the function.. `DeviceWiki` is supposed to be the root type returned by the api (even if now it's typed as `any`, it would be nice to autogenerate it in the future).. The child type is not exactly the same as `DeviceWiki` root and that's the reason why I'm not trying to reuse the function here...

Reply:
Ah thanks for the clarification, I didn't realize there was a distinction 👍🏻